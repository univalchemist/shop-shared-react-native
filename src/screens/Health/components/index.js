/* istanbul ignore file */

export {
  CallToActionLink,
  CardContainer,
  CardLoader,
  ErrorCard,
  TipCard,
} from './widgets';

export { default as ProductRecommendation } from './ProductRecommendation';
export { default as LifestyleTipsModal } from './LifestyleTipsModal';
export { default as LifestyleResults } from './LifestyleResults';
export { default as HealthScoreHistory } from './HealthScoreHistory';
export { default as HealthScore } from './HealthScore';
export { default as HealthResultCard } from './HealthResultCard';
export { default as GeneralTips } from './GeneralTips';
export { default as FaceAgingResults } from './FaceAgingResults';
