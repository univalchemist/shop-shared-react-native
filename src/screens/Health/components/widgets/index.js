export { default as CallToActionLink } from './CallToActionLink';
export { default as CardContainer } from './CardContainer';
export { default as CardLoader } from './CardLoader';
export { default as ErrorCard } from './ErrorCard';
export { default as TipCard } from './TipCard';
