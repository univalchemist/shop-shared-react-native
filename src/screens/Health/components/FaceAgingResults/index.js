import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Slider } from 'react-native-elements';
import {
  Box,
  Flex,
  Image,
  Loader,
  ScreenHeadingText,
  Text,
} from '@wrappers/components';
import theme from '@theme';
import PropTypes from 'prop-types';
import { useIntl } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard.js';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { categories, logAction } from '@store/analytics/trackingActions';

const styles = StyleSheet.create({
  container: { paddingHorizontal: 32, paddingBottom: 32 },
});

export const FaceAgingResults = ({ faceAging }) => {
  const intl = useIntl();
  const [selectedAgeIndex, setSelectedAgeIndex] = useState(0);
  return (
    <View style={styles.container}>
      {faceAging.isError ? (
        renderFaceAgingErrorPanel({ intl })
      ) : faceAging.faceAgingIsDone &&
        faceAging.expectedTotalResults === faceAging.currentTotalResults ? (
        renderFaceAgingResults(
          faceAging,
          selectedAgeIndex,
          setSelectedAgeIndex,
          intl,
        )
      ) : (
        <>
          <ScreenHeadingText mb={2}>
            {intl.formatMessage({ id: 'futureMeTitle' })}
          </ScreenHeadingText>
          <CardContainer
            theme={theme}
            justifyContent="center"
            color={theme.colors.white}
            flexGrow={1}
          >
            <Loader
              primary
              textAlign={'center'}
              loadingText={intl.formatMessage({
                id: 'health.healthScoreHistory.lifestyleImage.processing',
                defaultMessage:
                  'Analysing in progress. This may take a little while. Please do not navigate away.',
              })}
            />
          </CardContainer>
        </>
      )}
    </View>
  );
};

const renderFaceAgingResults = (
  faceAging,
  selectedAgeIndex,
  setSelectedAgeIndex,
  intl,
) => {
  let faceAges = [];
  if (faceAges.length === 0) {
    faceAges = Object.keys(faceAging.results).sort();
  }
  const selectedAge = faceAges[selectedAgeIndex];
  const selectedAgeForDisplay = selectedAge == '67' ? '65' : selectedAge;

  return faceAges.length > 0 ? (
    <>
      <ScreenHeadingText mb={2}>
        {intl.formatMessage({ id: 'futureMeTitle' })}
      </ScreenHeadingText>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          mt={8}
          mb={8}
        >
          {intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
        </Text>
        {renderFaceAgingImages(selectedAge, faceAging, intl)}
        <Box mt={16}>
          <Slider
            style={{ width: '100%', height: 40 }}
            step={1}
            minimumValue={0}
            maximumValue={faceAges.length - 1}
            thumbTintColor={theme.colors.black}
            minimumTrackTintColor={theme.colors.black}
            maximumTrackTintColor={theme.colors.gray[4]}
            onValueChange={faceAgeIndex => {
              setSelectedAgeIndex(faceAgeIndex);
            }}
            onSlidingComplete={() => {
              logAction({
                category: categories.LIFESTYLE_OVERVIEW,
                action: `Slider age ${selectedAgeForDisplay}`,
              });
            }}
          />
          <Flex flex={1} flexDirection="row" />
        </Box>
        <Text
          accessibilityLabel={intl.formatMessage({
            id: 'health.faceAgingDescription',
          })}
          accessible={true}
          fontSize={14}
          mt={8}
          mb={16}
        >
          {intl.formatMessage({ id: 'health.faceAgingDescription' })}
        </Text>
      </Box>
    </>
  ) : (
    <Box />
  );
};

const renderFaceAgingImages = (selectedAge, faceAging, intl) => {
  return (
    <>
      {Object.keys(faceAging.results).map((faceAge, index) => {
        return (
          <Flex
            key={index}
            flexDirection="row"
            style={faceAge === selectedAge ? {} : { display: 'none' }}
          >
            <Box flex={1} marginRight={8}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingHealthyLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].healthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingHealthyLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingHealthyLifestyle' })}
                </Text>
              </Box>
            </Box>
            <Box flex={1}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingCurrentLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].unhealthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingCurrentLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingCurrentLifestyle' })}
                </Text>
              </Box>
            </Box>
          </Flex>
        );
      })}
    </>
  );
};

const renderFaceAgingErrorPanel = ({ intl }) => (
  <Box>
    <ScreenHeadingText mb={2}>
      {intl.formatMessage({ id: 'futureMeTitle' })}
    </ScreenHeadingText>
    <ErrorCard color={theme.colors.white} />
  </Box>
);

const mapStateToProps = ({ health: { faceAging } }) => ({
  faceAging,
});

const enhance = compose(connect(mapStateToProps));

export default enhance(FaceAgingResults);

FaceAgingResults.propTypes = {
  faceAging: PropTypes.shape({
    faceAgingIsDone: PropTypes.bool,
    expectedTotalResults: PropTypes.number,
    currentTotalResults: PropTypes.number,
    results: PropTypes.shape({}),
  }),
};
