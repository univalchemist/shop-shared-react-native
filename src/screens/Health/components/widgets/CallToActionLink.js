import React from 'react';
import { Text, Flex, Image, Divider } from '@wrappers/components';
import styled from 'styled-components/native';
import theme from '@theme';

const StyledText = styled(Text)`
  color: ${theme.colors.fonts.blackLink};
`;

const StyledImage = styled(Image)`
  width: 16px;
  height: 16px;
  align-self: center;
  margin-left: 10;
`;

export const CallToActionLink = ({ text, icon }) => {
  return (
    <Flex height={54} flexDirection="column">
      <Divider full={true} />
      <Flex
        flexDirection="row"
        paddingLeft={24}
        paddingRight={24}
        paddingTop={16}
      >
        <StyledText>{text}</StyledText>
        {icon && <StyledImage source={icon} />}
      </Flex>
    </Flex>
  );
};
