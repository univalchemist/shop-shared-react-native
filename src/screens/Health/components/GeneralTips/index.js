import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { GeneralTipsCarousel } from './GeneralTipsCarousel';
import { GeneralTipsLoader } from './GeneralTipsLoader';
import { PromiseStatus } from '@middlewares';
import { Dimensions } from 'react-native';
import { useTheme } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';
import { Box, Text } from '@wrappers/components';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { FormattedMessage } from 'react-intl';
import { fetchLifestyleTips } from '@store/health/actions';

export const GeneralTips = ({ lifestyleTips, fetchLifestyleTips, locale }) => {
  const theme = useTheme();
  const horizontalMarginBetweenCards = theme.space[3];
  const viewportWidth = Dimensions.get('window').width;
  const viewportPadding = theme.space[4];
  const cardWidth = viewportWidth - viewportPadding * 2;
  const cardWidthWithSpacing = cardWidth + horizontalMarginBetweenCards;
  const cardHeight = 252.5;
  const viewportPaddingInsideCard = viewportPadding + 24;
  const loaderWidthInsideCard = viewportWidth - viewportPaddingInsideCard * 2;

  useEffect(() => {
    fetchLifestyleTips();
  }, [locale]);

  const renderGeneralTips = () => {
    switch (lifestyleTips.status) {
      case PromiseStatus.SUCCESS: {
        return lifestyleTips.tips.general ? (
          <GeneralTipsCarousel
            data={lifestyleTips.tips.general}
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
          />
        ) : (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      case PromiseStatus.ERROR: {
        return (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      default: {
        return (
          <GeneralTipsLoader
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
            loaderWidthInsideCard={loaderWidthInsideCard}
          />
        );
      }
    }
  };

  return (
    <>
      <Box>
        <Text
          color="gray.0"
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          ml={32}
          mt={40}
        >
          <FormattedMessage id="health.sectionTitle.generalTips" />
        </Text>
      </Box>
      {renderGeneralTips()}
    </>
  );
};

const mapStateToProps = ({ health: { lifestyleTips }, intl: { locale } }) => ({
  lifestyleTips,
  locale,
});
const mapDispatchToProps = { fetchLifestyleTips };
const enhance = compose(connect(mapStateToProps, mapDispatchToProps));

GeneralTips.propTypes = {
  lifestyleTips: PropTypes.shape({
    tips: PropTypes.shape({ general: PropTypes.array }),
    status: PropTypes.string,
  }),
};

export default enhance(GeneralTips);
