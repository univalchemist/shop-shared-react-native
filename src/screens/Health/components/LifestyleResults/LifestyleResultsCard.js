import React from 'react';
import { Box, Flex, Image, Text } from '@wrappers/components';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { LIFESTYLE_TIPS_MODAL } from '@routes';
import { TouchableHighlight } from 'react-native';
import { CallToActionLink } from '@screens/Health/components/widgets/CallToActionLink';
import { categories, logAction } from '@store/analytics/trackingActions';

export const Container = styled(TouchableHighlight)`
  background-color: ${props => props.theme.colors.white};
  border-top-color: ${props => props.color};
  flex-grow: 1;
  border-top-width: 4;
  border-radius: 4;
  padding-top: 28;
  display: flex;
  shadow-offset: 0px 2px;
  shadow-opacity: 0.05px;
  shadow-radius: 3;
  shadow-color: #000;
`;

export const Score = styled(Text)`
  color: ${props => props.color};
  font-size: ${props => props.theme.fontSizes[5]};
  font-weight: ${props => props.theme.fontWeights.bolder};
  line-height: ${props => props.theme.lineHeights[5]};
  letter-spacing: -1.5;
  margin-top: 6;
  margin-right: 30;
`;

const Title = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
`;

const Status = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  font-weight: ${props => props.theme.fontWeights.bold};
  letter-spacing: 0.3;
`;

const Description = styled(Text)`
  padding-top: 32;
  color: ${props => props.theme.colors.gray[8]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 24;
  letter-spacing: 0.3;
  padding-bottom: 32;
`;

const LifestyleResultsCard = ({
  color,
  score,
  title,
  description,
  status,
  iconImage,
  tips,
  textColor,
  navigation,
}) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <Container
      accessible={true}
      accessibilityRole="button"
      underlayColor={theme.colors.touchableOverlayColor}
      theme={theme}
      color={color}
      onPress={() => {
        logAction({
          category: categories.LIFESTYLE_OVERVIEW,
          action: `View result ${tips.category}`,
        });
        navigation.navigate(LIFESTYLE_TIPS_MODAL, { tips });
      }}
    >
      <>
        <Flex flexGrow={1} flexDirection={'row'} px={24}>
          <Box alignSelf={'center'} marginRight={30}>
            {iconImage ? (
              <Image
                accessible={true}
                accessibilityLabel={intl.formatMessage({
                  id: 'health.accessibilityLabel.icon',
                })}
                source={iconImage}
              />
            ) : (
              <Score
                accessible={true}
                accessibilityLabel={score.toFixed(1)}
                color={textColor}
              >
                {score.toFixed(1)}
              </Score>
            )}
          </Box>
          <Flex flexGrow={1} flexDirection={'column'}>
            <Title accessible={true} accessibilityLabel={title} flexGrow={1}>
              {title}
            </Title>
            <Status accessible={true} accessibilityLabel={status} flexGrow={1}>
              {status}
            </Status>
          </Flex>
          <Description
            accessible={true}
            accessibilityLabel={description}
            flexGrow={1}
          >
            {description}
          </Description>
        </Flex>
        <CallToActionLink
          text={intl.formatMessage({
            id: 'health.LifestyleTips.ViewLifestyleTips',
          })}
        />
      </>
    </Container>
  );
};

LifestyleResultsCard.propTypes = {
  color: PropTypes.string.isRequired,
  score: PropTypes.number,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  tips: PropTypes.shape({
    category: PropTypes.string,
    result: PropTypes.string,
    tipDescription: PropTypes.string,
  }),
};

export default LifestyleResultsCard;
