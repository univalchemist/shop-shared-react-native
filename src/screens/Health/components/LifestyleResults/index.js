import { LifestyleResultsCarousel } from './LifestyleResultsCarousel';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Dimensions } from 'react-native';
import { Box, Text } from '@wrappers/components';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import { LifestyleResultsLoader } from './LifestyleResultsLoader';
import { PromiseStatus } from '@middlewares';
import { getLifestyleResults } from './util';
import { injectIntl, FormattedMessage } from 'react-intl';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';

export const LifestyleResults = ({
  lifestyleResults,
  fetchUserLifestyleResultsStatus,
  navigation,
}) => {
  const theme = useTheme();
  const intl = useIntl();
  const horizontalMarginBetweenCards = theme.space[3];
  const viewportWidth = Dimensions.get('window').width;
  const viewportPadding = theme.space[4];
  const cardWidth = viewportWidth - viewportPadding * 2;
  const cardWidthWithSpacing = cardWidth + horizontalMarginBetweenCards;
  const cardHeight = 252.5;

  const renderLifestyleResults = () => {
    switch (fetchUserLifestyleResultsStatus) {
      case PromiseStatus.SUCCESS: {
        const lifestyleResultsArray = getLifestyleResults(
          lifestyleResults,
          theme,
          intl,
        );

        return (
          <LifestyleResultsCarousel
            data={lifestyleResultsArray}
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
            navigation={navigation}
          />
        );
      }
      case PromiseStatus.ERROR: {
        return (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={theme.colors.error[3]} />
          </Box>
        );
      }
      default: {
        return (
          <LifestyleResultsLoader
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
          />
        );
      }
    }
  };

  return (
    <>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.sectionTitle.lifestyleResults',
          })}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          ml={32}
          mt={40}
        >
          <FormattedMessage id="health.sectionTitle.lifestyleResults" />
        </Text>
      </Box>
      {renderLifestyleResults()}
    </>
  );
};

const mapStateToProps = ({
  health: { lifestyleResults, fetchUserLifestyleResultsStatus },
}) => ({
  lifestyleResults,
  fetchUserLifestyleResultsStatus,
});

const enhance = compose(connect(mapStateToProps, {}), injectIntl);

LifestyleResults.propTypes = {
  fetchUserLifestyleResultsStatus: PropTypes.string,
  lifestyleResults: PropTypes.shape({
    bmiScore: PropTypes.number,
    bmiClass: PropTypes.string,
  }),
};

export default enhance(LifestyleResults);
