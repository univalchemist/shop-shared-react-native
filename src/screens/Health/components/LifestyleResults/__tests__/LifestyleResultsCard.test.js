import React from 'react';
import LifestyleResultsCard, {
  Container,
  Score,
} from '../LifestyleResultsCard';
import { renderForTest } from '@testUtils';
import { Image } from '@wrappers/components';
import { diabetesRiskGreen } from '@images';
import { act, fireEvent } from 'react-native-testing-library';
import { LIFESTYLE_TIPS_MODAL } from '@routes';
import { CallToActionLink } from '../../widgets/CallToActionLink';
import messages from '@messages/en-HK';

const navigation = {
  navigate: jest.fn(),
};

describe('LifestyleResultsCard', () => {
  const color = 'red';
  const score = 10;
  const title = 'title';
  const description = 'description';
  const status = 'status';
  const textColor = '#333333';

  let lifestyleResultsCard;
  let lifestyleCard;
  beforeEach(() => {
    lifestyleResultsCard = renderForTest(
      <LifestyleResultsCard
        color={color}
        score={score}
        title={title}
        description={description}
        status={status}
        textColor={textColor}
        navigation={navigation}
      />,
    );
    lifestyleCard = lifestyleResultsCard.queryAllByType(Container);
  });

  it('should render snapshot', () => {
    expect(lifestyleResultsCard.toJSON()).toMatchSnapshot();
  });

  it('shoud render a lifestyle card', () => {
    expect(lifestyleCard.length).toEqual(1);
  });

  it('should render the correct color', () => {
    expect(lifestyleCard[0].props.color).toEqual(color);
  });

  it('should render title, status and description', () => {
    expect(lifestyleResultsCard.queryAllByText(title).length).toEqual(1);
    expect(lifestyleResultsCard.queryAllByText(status).length).toEqual(1);
    expect(lifestyleResultsCard.queryAllByText(description).length).toEqual(1);
  });

  it('should render score to be 10.0', () => {
    const score = lifestyleResultsCard.queryAllByType(Score)[0];
    expect(score.props.children).toEqual('10.0');
  });

  it('should render correct score when iconImage prop is not passed', () => {
    expect(lifestyleResultsCard.queryAllByType(Score).length).toEqual(1);
  });

  it('should not render icon when iconImage prop is not passed', () => {
    const image = lifestyleResultsCard.queryAllByType(Image);

    expect(image.length).toBe(0);
  });

  it('should render icon when iconImage prop is passed', () => {
    const lifestyleResultsCard = renderForTest(
      <LifestyleResultsCard
        color={'red'}
        score={22.7}
        title={'title'}
        description={'Overweight'}
        status={'description really really long'}
        iconImage={diabetesRiskGreen}
        navigation={navigation}
      />,
    );

    const image = lifestyleResultsCard.queryAllByType(Image);
    const score = lifestyleResultsCard.queryAllByType(Score);

    expect(image.length).toBe(1);
    expect(score.length).toBe(0);
    expect(image[0].props.source).toBe(diabetesRiskGreen);
  });

  it('should invoke the callback when clicked', () => {
    const color = 'red';
    const score = 22.7;
    const title = 'title';
    const status = 'Overweight';
    const description = 'description really really long';
    const textColor = '#333333';

    const tips = { category: '', result: '', tipDescription: '' };
    const component = renderForTest(
      <LifestyleResultsCard
        color={color}
        score={score}
        title={title}
        description={description}
        status={status}
        tips={tips}
        textColor={textColor}
        navigation={navigation}
      />,
    );

    const lifestyleCard = component.queryAllByType(Container);
    act(() => {
      fireEvent.press(lifestyleCard[0]);
    });

    expect(navigation.navigate).toHaveBeenCalledWith(LIFESTYLE_TIPS_MODAL, {
      tips,
    });
  });

  it('should render callToActionLink', () => {
    const lifestyleResultsCard = renderForTest(
      <LifestyleResultsCard
        color={'red'}
        score={22.7}
        title={'title'}
        description={'Overweight'}
        status={'description really really long'}
        iconImage={diabetesRiskGreen}
        navigation={navigation}
      />,
    );

    const callToActionLink = lifestyleResultsCard.queryAllByType(
      CallToActionLink,
    );

    expect(callToActionLink.length).toBe(1);
    expect(callToActionLink[0].props.text).toBe(
      messages['health.LifestyleTips.ViewLifestyleTips'],
    );
    expect(callToActionLink[0].props.icon).toBe(undefined);
  });
});
