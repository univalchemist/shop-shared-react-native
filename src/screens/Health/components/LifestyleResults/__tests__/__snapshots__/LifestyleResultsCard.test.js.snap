// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifestyleResultsCard should render snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderRadius": 4,
                              "borderTopColor": "red",
                              "borderTopWidth": 4,
                              "display": "flex",
                              "flexGrow": 1,
                              "paddingTop": 28,
                              "shadowColor": "#000",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.05,
                              "shadowRadius": 3,
                            },
                          ]
                        }
                      >
                        <View
                          flexDirection="row"
                          flexGrow={1}
                          px={24}
                          style={
                            Array [
                              Object {
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "paddingLeft": 24,
                                "paddingRight": 24,
                              },
                            ]
                          }
                        >
                          <View
                            alignSelf="center"
                            marginRight={30}
                            style={
                              Array [
                                Object {
                                  "alignSelf": "center",
                                  "marginRight": 30,
                                },
                              ]
                            }
                          >
                            <Text
                              accessibilityLabel="10.0"
                              accessible={true}
                              color="#333333"
                              fontFamily="default"
                              fontSize={16}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#333333",
                                    "fontFamily": "Arial",
                                    "fontSize": 32,
                                    "fontWeight": "800",
                                    "letterSpacing": -1.5,
                                    "lineHeight": 37,
                                    "marginRight": 30,
                                    "marginTop": 6,
                                  },
                                ]
                              }
                            >
                              10.0
                            </Text>
                          </View>
                          <View
                            flexDirection="column"
                            flexGrow={1}
                            style={
                              Array [
                                Object {
                                  "display": "flex",
                                  "flexDirection": "column",
                                  "flexWrap": "wrap",
                                },
                              ]
                            }
                          >
                            <Text
                              accessibilityLabel="title"
                              accessible={true}
                              color="gray.1"
                              flexGrow={1}
                              fontFamily="default"
                              fontSize={16}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#212121",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              title
                            </Text>
                            <Text
                              accessibilityLabel="status"
                              accessible={true}
                              color="gray.1"
                              flexGrow={1}
                              fontFamily="default"
                              fontSize={16}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#212121",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "fontWeight": "700",
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              status
                            </Text>
                          </View>
                          <Text
                            accessibilityLabel="description"
                            accessible={true}
                            color="gray.1"
                            flexGrow={1}
                            fontFamily="default"
                            fontSize={16}
                            lineHeight={22}
                            style={
                              Array [
                                Object {
                                  "color": "#666666",
                                  "fontFamily": "Arial",
                                  "fontSize": 16,
                                  "letterSpacing": 0.3,
                                  "lineHeight": 24,
                                  "paddingBottom": 32,
                                  "paddingTop": 32,
                                },
                              ]
                            }
                          >
                            description
                          </Text>
                        </View>
                        <View
                          flexDirection="column"
                          height={54}
                          style={
                            Array [
                              Object {
                                "display": "flex",
                                "flexDirection": "column",
                                "flexWrap": "wrap",
                                "height": 54,
                              },
                            ]
                          }
                        >
                          <View
                            backgroundColor="gray.5"
                            height={1}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#e3e3e3",
                                  "height": 1,
                                  "width": "100%",
                                },
                              ]
                            }
                            width="100%"
                          />
                          <View
                            flexDirection="row"
                            paddingLeft={24}
                            paddingRight={24}
                            paddingTop={16}
                            style={
                              Array [
                                Object {
                                  "display": "flex",
                                  "flexDirection": "row",
                                  "flexWrap": "wrap",
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              color="gray.1"
                              fontFamily="default"
                              fontSize={16}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#333333",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              View lifestyle tips
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
